text/3-entrees/fixed-point.md

```diff
@@ -0,0 +1,327 @@
+
+```rust, ignore
+fn deposit_discrete(origin, val_to_add: u64) -> DispatchResult {
+	let _ = ensure_signed(origin)?;
+
+	let old_value = DiscreteAccount::get();
+
+	// Update storage for discrete account
+	DiscreteAccount::put(old_value + val_to_add);
+
+	// Emit event
+	Self::deposit_event(Event::DepositedDiscrete(val_to_add));
+	Ok(())
+}
+```
+
+The flow of these deposit and withdraw extrinsics is entirely straight-forward. They each perform a simple addition or substraction from the stored value, and they have nothing to do with interest.
+
+Because the interest is paid discretely every ten blocks it can be handled independently of deposits and withdrawals. The interest calculation happens automatically in the `on_finalize` block.
+
+```rust, ignore
+fn on_finalize(n: T::BlockNumber) {
+	// Apply newly-accrued discrete interest every ten blocks
+	if (n % 10.into()).is_zero() {
+
+		// Calculate interest Interest = principle * rate * time
+		// We can use the `*` operator for multiplying a `Percent` by a u64
+		// because `Percent` implements the trait Mul<u64>
+		let interest = Self::discrete_interest_rate() * DiscreteAccount::get() * 10;
+
+		// The following line, although similar, does not work because
+		// u64 does not implement the trait Mul<Percent>
+		// let interest = DiscreteAccount::get() * Self::discrete_interest_rate() * 10;
+
+		// Update the balance
+		let old_balance = DiscreteAccount::get();
+		DiscreteAccount::put(old_balance + interest);
+
+		// Emit the event
+		Self::deposit_event(Event::DiscreteInterestApplied(interest));
+	}
+}
+```
+
+`on_finalize` is called at the end of every block, but we only want to pay interest every ten blocks, so the first thing we do is check whether this block is a multiple of ten. If it is we calculate the interest due by the formula `interest = principle * rate * time`. As the comments explain, there is some subtlety in the order of the multiplication. You can multiply `PerCent * u64` but not `u64 * PerCent`.
+
+
+### Continuously Compounding
+
+You can imagine increasing the frequency a which the interest is paid out. Increasing the frequency enough approaches [continuously compounding interest](https://en.wikipedia.org/wiki/Compound_interest#Continuous_compounding). Calculating continuously compounding interest requires the [exponential function](https://en.wikipedia.org/wiki/Exponential_function) which is not available using Substrate's `PerThing` types. Luckily exponential and other [transcendental functions](https://en.wikipedia.org/wiki/Transcendental_function) are available in substrate-fixed, which is why we've chosen to use it for this example.
+
+With continuously compounded interest, we _could_ update the interest in `on_finalize` as we did before, but it woudl need to be updated every single block. Instead we wait until a user tries to use the account (to deposit or withdraw funds), and then calculate the account's current value "just in time".
```

Comment:
```suggestion
With continuously compounded interest, we _could_ update the interest in `on_finalize` as we did before, but it would need to be updated every single block. Instead we wait until a user tries to use the account (to deposit or withdraw funds), and then calculate the account's current value "just in time".
```